docker-entrypoint.sh-rootless

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+CONFIG_FILE=/data/lldap_config.toml
+
+if [[ ( ! -w "/data" ) ]] || [[ ( ! -d "/data" ) ]]; then
```

Comment:
For compatibility, we can use `[` instead of `[[`, and I don't think we need the inner parentheses.

Reply:
Also, we don't really need `/data` to be writable: it's only used to set up the initial config.

How about:
- If the config file is missing:
  - if the folder is writable, copy the template
  - otherwise, error.

Reply:
How just just check if config file exist? In the end if it fail likely and mostly is permission issue.
```
lldap-lldap-1  | [entrypoint] Copying the default config to /data/lldap_config.toml
lldap-lldap-1  | [entrypoint] Edit this /data/lldap_config.toml to configure LLDAP.
lldap-lldap-1  | cp: can't create '/data/lldap_config.toml': Permission denied
lldap-lldap-1 exited with code 1
```

Reply:
Yeah, but the error is not as nice. You can say "Could not create the default config file /data/lldap_config.toml, please copy the template from the repository to get started" or something similar.

Reply:
Something like this I think sufficient?
```
#!/usr/bin/env bash
set -euo pipefail

CONFIG_FILE=/data/lldap_config.toml

if [ ! -f "$CONFIG_FILE" ]; then
  echo "[entrypoint] Copying the default config to $CONFIG_FILE"
  echo "[entrypoint] Edit this $CONFIG_FILE to configure LLDAP."
  if cp /app/lldap_config.docker_template.toml $CONFIG_FILE; then
     echo "Configuration copied successfully."
  else
     echo "Fail to copy configuration, check permission on /data or manually create one by copying from LLDAP repository"
     exit 1
  fi
fi

echo "> Starting lldap.."
echo ""
exec /app/lldap "$@"
exec "$@"
```

Reply:
Looks good!

Reply:
Oh, on my testing I saw something weird, maybe just a nitpick
```
Attaching to lldap-lldap-1
lldap-lldap-1  | [entrypoint] Copying the default config to /data/lldap_config.toml
lldap-lldap-1  | [entrypoint] Edit this /data/lldap_config.toml to configure LLDAP.
lldap-lldap-1  | cp: can't create '/data/lldap_config.toml': Permission denied
lldap-lldap-1  | Fail to copy configuration, check permission on /data or manually create one by copying from LLDAP repository
####I Forgot to Break Here ####
lldap-lldap-1  | > Starting lldap..
lldap-lldap-1  | 
###################
# Why LLDAP still tell load from /data/lldap_config.toml, as `/data/lldap_config.toml` non-exist.
###################
lldap-lldap-1  | Loading configuration from /data/lldap_config.toml
lldap-lldap-1  | Configuration: Configuration {
lldap-lldap-1  |     ldap_host: "0.0.0.0",
lldap-lldap-1  |     ldap_port: 3890,
lldap-lldap-1  |     http_host: "0.0.0.0",
lldap-lldap-1  |     http_port: 17170,
lldap-lldap-1  |     jwt_secret: ***SECRET***,
lldap-lldap-1  |     ldap_base_dn: "dc=example,dc=com",
lldap-lldap-1  |     ldap_user_dn: UserId(
lldap-lldap-1  |         CaseInsensitiveString(
lldap-lldap-1  |             "admin",
lldap-lldap-1  |         ),
lldap-lldap-1  |     ),


```